### PR TITLE
[ShEIP-91] Unit tests added for campaign pages and components

### PR DESCRIPTION
**Pull request by akhan-mozilla**
## References

[SHEP-91](https://mozilla-hub.atlassian.net/browse/SHEP-91)

## Problem Statement

We need to add unit tests for the campaign pages and their components now that the Jest setup has been completed and merged.

## Proposed Changes

1. Added unit tests for the campaign pages and their components.
2. Set the threshold value for Jest test coverage to 80%.

## Verification Steps

Run the `make test-react` command for verification

## Visuals
Screenshot 2024-10-11 at 12 34 25 PM

[SHEP-91]: https://mozilla-hub.atlassian.net/browse/SHEP-91?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ad-ops-dashboard/src/server.ts
Comment (nanagb) [inline ad-ops-dashboard/src/server.ts:1]: NIT: This server functions and other related ones tied to msw are triggering eslint errors related to using `any` I think we have a story for this, so mainly flagging.

File: ad-ops-dashboard/src/components/Forms/CampaignForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import "@testing-library/jest-dom";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import CampaignForm from "./CampaignForm";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { useCreateCampaignMutation } from "../../data/campaigns";
+import { defaultCampaign, newCampaign } from "../../fixtures/campaignFixtures";
+
+jest.mock("../../data/campaigns", () => ({
+  useCreateCampaignMutation: jest.fn(),
+  useUpdateCampaignMutation: jest.fn(),
```
Comment (nanagb) [inline ad-ops-dashboard/src/components/Forms/CampaignForm.test.tsx:10]: I noticed this is defined but not being used. Will you be adding more tests later on?

Comment (akhan-mozilla) [reply ad-ops-dashboard/src/components/Forms/CampaignForm.test.tsx:10]: The `useCreateCampaignMutation` is used in the Create Campaign[ test case ](https://github.com/mozilla-services/consvc-shepherd/blob/f76a0068e109a2cf346669d4eaeb1f8014d1a701/ad-ops-dashboard/src/components/Forms/CampaignForm.test.tsx#L47) to ensure that the [mutation](https://github.com/mozilla-services/consvc-shepherd/blob/f76a0068e109a2cf346669d4eaeb1f8014d1a701/ad-ops-dashboard/src/components/Forms/CampaignForm.test.tsx#L112) is called during campaign creation. The `useUpdateCampaignMutation` has been mocked to ensure that the actual `[useUpdateCampaignMutation](https://github.com/mozilla-services/consvc-shepherd/blob/f76a0068e109a2cf346669d4eaeb1f8014d1a701/ad-ops-dashboard/src/components/Forms/CampaignForm.tsx#L52)` is not invoked from the CampaignForm component.

**Review by nanagb — APPROVED** [review]
Thanks for getting these all setup @akhan-mozilla this is a solid and extensive amount of test setup and it provides a a helpful template for the rest of the team to use for creating new tests.

**Pull request merged**